fastlane/fastlane
# [core] Skip repeating the output again when an sh command fails

### Checklist

### Motivation and Context

Resolves #19416
Resolves #11140

### Description


I'm changing `sh` *not* to repeat the command output again when an error happens.
On output-heavy shell calls (in our case running Unity) this causes the entire output of Unity being repeated for absolutely no gain.
To make matters worse, without passing an `error_callback` there's even an exception raised that `bundler` picks up then and decides to repeat the output yet again. Yikes. :)

I've tested my changes both by running the rspec tests (which is also the reason why I had to change the tests) and by calling a script that fails through `sh` from my own project.

### Testing Steps


Simply call any script that should fail and observe the output only appearing once.
